src/include/rimage/rimage.h

```diff
@@ -138,6 +138,11 @@ struct fw_image_manifest_module {
 	struct sof_man_module *mod_man;
 };
 
+enum rtos_variant {
+	RTOS_XTOS,
```

Comment:
Rather than call this Zephyr/xtos and be RTOS specific, this is really about how we handle cache memory mappings. Can we name this to refer to the different ways we handle memory.

Reply:
@lgirdwood well, at a low level that's about how linker scripts are written. Zephyr linker scripts use cached aliases for certain sections and uncached for others. Our XTOS linker scripts don't do that, all sections there have cached addresses. I thought it could be useful for rimage to know what kind of an image it's handling, not only for address mangling. But we can make it a memory type flag too, sure.
But in general, in principle it should be possible to always use the "Zephyr version" - to always convert addresses. Converting cached aliases to cached again shouldn't change them. I'm testing that version now too #149 + https://github.com/thesofproject/sof/pull/7226 - if that works, maybe we need the first patch in this PR